server/gameconnection.py

```diff
@@ -143,16 +166,20 @@ def player(self, val: Player):
         Connect two peers
         :return: None
         """
-        await self.send_ConnectToPeer(
-            player_name=peer.player.login,
-            player_uid=peer.player.id,
-            offer=True
-        )
-        await peer.send_ConnectToPeer(
-            player_name=self.player.login,
-            player_uid=self.player.id,
-            offer=False
-        )
+        if peer:
+            await self.send_ConnectToPeer(
+                player_name=peer.player.login,
+                player_uid=peer.player.id,
+                offer=True
+            )
+
+        if peer:
```

Comment:
Hmm, now that I think about it I'm actually not sure how passing weak references to functions interact. 

The intent is that we need to check the existence of the reference before using it, otherwise for example `peer.player` could throw a `NoneType` error. And it is also the case that whenever you `await`, you need to re-check all of your weak references again since they could have been deleted and cleaned up by the GC since your first check. Now the only thing I'm not sure is how this interaction works when your weak reference is passed as a function argument. I would assume that the function argument its self would still be a weak reference and you'd have to check it every time you `await`.